systest/tests/common.go

```diff
@@ -196,7 +195,12 @@ func layersStream(
 		if ok && s.Code() != codes.OK {
 			logger.Warn("layers stream error", zap.String("client", node.Name), zap.Error(err), zap.Any("status", s))
 			if s.Code() == codes.Unavailable {
-				return nil
+				if retries == attempts {
+					return errors.New("layer stream unavailable")
+				}
+				retries++
+				time.Sleep(retryBackoff)
+				goto BACKOFF
```

Comment:
I think we should avoid implement this ourselves. `grpc.NewClient` can be used with `grpc.WithBackoffConfig()`. You could add that to `ensurePubCon` and `ensurePrivCon` in `node.go` and not need to use this loop to retry automatically. 🙂

Reply:
I was digging through the grpc code and the only thing I found this to be relevant to is establishing connections. Looking at their documentation, it appears that `Unavailable` is not related to establishing connections, but rather happens later in the flow. I actually doubt they would do blocking retries for application-level RPC calls, this seems out of scope for an RPC library (it is up to the application to decide whether and how to retry).